File: app/src/main/java/cn/ifafu/ifafu/mvp/login/LoginContract.java
package cn.ifafu.ifafu.mvp.login;

import androidx.annotation.DrawableRes;

import cn.ifafu.ifafu.data.entity.Response;
import cn.ifafu.ifafu.data.entity.User;
import cn.ifafu.ifafu.mvp.base.i.IView;
import cn.ifafu.ifafu.mvp.base.i.IZFModel;
import cn.ifafu.ifafu.mvp.base.i.IZFPresenter;
import io.reactivex.Observable;

public class LoginContract {

    public interface View extends IView {

        String getAccountText();

        String getPasswordText();

        void showCloseBtn();

        void setBackgroundLogo(@DrawableRes int resId);
    }

    public interface Presenter extends IZFPresenter {

        void onLogin();

        void checkAccount(String account);

    }

    public interface Model extends IZFModel {

        Observable<Response<String>> login(User user);

        void saveUser(User user);
    }

}

File: app/src/main/java/cn/ifafu/ifafu/mvp/login/LoginContract.kt
package cn.ifafu.ifafu.mvp.login

import androidx.annotation.DrawableRes
import cn.ifafu.ifafu.base.i.IView
import cn.ifafu.ifafu.base.ifafu.IZFModel
import cn.ifafu.ifafu.base.ifafu.IZFPresenter
import cn.ifafu.ifafu.data.entity.Response
import cn.ifafu.ifafu.data.entity.User
import io.reactivex.Observable

class LoginContract {

    interface View : IView {

        fun getAccountText(): String

        fun getPasswordText(): String

        fun showCloseBtn()

        fun setBackgroundLogo(@DrawableRes resId: Int)
    }

    interface Presenter : IZFPresenter {

        fun onLogin()

        fun checkAccount(account: String)

    }

    interface Model : IZFModel {

        fun login(user: User): Observable<Response<String>>

        fun saveUser(user: User)
    }

}
